### PR TITLE
Hook palette to manual clean

### PR DESCRIPTION
**Pull request by csbailey5t**

### CODE REVIEW

File: app/controllers/collection_templates/build_controller.rb [hunk 1]
```diff
@@ -24,7 +24,13 @@ def update
     when 'auto_clean'
       ImageEnhanceJob.new.perform(@collection_template)
     when 'image_clean'
+      collection = Collection.find(@collection_template.collection_id)
```
Comment (mejackreed) [inline app/controllers/collection_templates/build_controller.rb]: I think you could probably shorten this to: `@collection_template.collection` no?

Comment (mejackreed) [reply app/controllers/collection_templates/build_controller.rb]: And probably remove this temp variable

File: app/jobs/create_palette_job.rb [hunk 2]
```diff
@@ -5,9 +5,11 @@
 class CreatePaletteJob < ApplicationJob
   queue_as :default
 
-  def perform(collection)
+  def perform(palette_params, collection)
```
Comment (mejackreed) [inline app/jobs/create_palette_job.rb:8]: where were `palette_params` coming from before?

File: app/models/collection_template.rb [hunk 3]
```diff
@@ -54,16 +55,35 @@ def image_clean_to_formal_json
...
 
+  def posterize_params
+    json_params = HashWithIndifferentAccess.new(image_clean)
+    "-p '#{palette}' -m #{json_params['posterize_method']} " \
+      "#{json_params['posterize']}"
+  end
+
+  def palette
+    collection = Collection.find(collection_id)
+    collection.palette
```
Comment (mejackreed) [inline app/models/collection_template.rb]: You could just use: `collection.palette` here

Also perhaps a more Ruby/Railsy way would be at the top of the file:

```
delegate : palette, to: :collection
```

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -12,6 +12,9 @@
 # Add additional requires below this line. Rails is not loaded until this point!
 
 Capybara.javascript_driver = :poltergeist
+Capybara.register_driver :poltergeist do |app|
+  Capybara::Poltergeist::Driver.new(app, timeout: 1.minute)
```
Comment (mejackreed) [inline spec/rails_helper.rb]: Are there specific tests that need this timeout? Might be good to just put the timeout on the tests that need it rather than everywhere. That way you know which ones are long running.

**Review by mejackreed — COMMENTED** [review]
A few comments, just my 2 cents. Looks great overall

File: app/jobs/create_palette_job.rb (hunk 2 above)
Comment (csbailey5t) [reply app/jobs/create_palette_job.rb:8]: Before, we simply didn't pass any palette_params. We were going entirely with default. I added these in now since we need to re-calculate the palette based on the params from the posterize chunk of the manual clean form.

File: spec/rails_helper.rb (hunk 4 above)
Comment (csbailey5t) [reply spec/rails_helper.rb]: There were three specific feature tests that were timing out due to the palette generation.I'll figure out how to change the timeout just for those tests.

**Review by mejackreed — APPROVED** [review]
👍
